### PR TITLE
Add strict type on Notifications tests

### PR DESCRIPTION
**Pull request by skjnldsv**
Followup https://github.com/nextcloud/server/pull/15912#issuecomment-501210838

### CODE REVIEW

**Comment by georgehrke** [comment]
I think we should add strict return types as well. :)

**Comment by skjnldsv** [comment]
> I think we should add strict return types as well. :)

Do_it_shia_laboeuf.gif

File: lib/public/Notification/INotification.php [hunk 1]
```diff
@@ -80,7 +80,7 @@ public function getDateTime();
 	 * @throws \InvalidArgumentException if the object type or id is invalid
 	 * @since 9.0.0
 	 */
-	public function setObject($type, $id);
+	public function setObject(string $type, $id);
```
Comment (ChristophWurst) [inline lib/public/Notification/INotification.php]: will do :boom: if you don't adjust the notifications app accordingly

Comment (skjnldsv) [reply lib/public/Notification/INotification.php]: Well, it was throwing for the same check anyway?
Why would it matter than it throw a TypeError or InvalidArgumentException? :thinking:

Comment (ChristophWurst) [reply lib/public/Notification/INotification.php]: nevermind. the implementation is also here in server, right? then it's fine

**Comment by juliushaertl** [comment]
Two remaining :wink: 

```
1) OCA\Comments\Tests\Unit\Controller\NotificationsTest::testViewSuccess
133 | TypeError: Argument 1 passed to Mock_INotification_fc0f9ea9::setUser() must be of the type string, null given, called in /drone/src/apps/comments/lib/Controller/Notifications.php on line 145
134 |  
135 | /drone/src/apps/comments/lib/Controller/Notifications.php:145
136 | /drone/src/apps/comments/lib/Controller/Notifications.php:118
137 | /drone/src/apps/comments/tests/Unit/Controller/NotificationsTest.php:153
138 |  
139 | 2) OCA\Comments\Tests\Unit\Controller\NotificationsTest::testViewNoFile
140 | TypeError: Argument 1 passed to Mock_INotification_fc0f9ea9::setUser() must be of the type string, null given, called in /drone/src/apps/comments/lib/Controller/Notifications.php on line 145
141 |  
142 | /drone/src/apps/comments/lib/Controller/Notifications.php:145
143 | /drone/src/apps/comments/lib/Controller/Notifications.php:118
144 | /drone/src/apps/comments/tests/Unit/Controller/NotificationsTest.php:216
145
```

**Comment by skjnldsv** [comment]
Done!

**Review by juliusknorr — APPROVED** [review]

**Review by ChristophWurst — APPROVED** [review]

**Comment by blizzz** [comment]
Also needs backports, doesn't it ?

**Comment by skjnldsv** [comment]
Yes!

**Pull request merged**